src/app/games/martian-chess/MartianChessRules.ts

```diff
@@ -0,0 +1,192 @@
+import { Coord } from 'src/app/jscaip/Coord';
+import { Direction, Vector } from 'src/app/jscaip/Direction';
+import { MGPNode } from 'src/app/jscaip/MGPNode';
+import { Player } from 'src/app/jscaip/Player';
+import { GameStatus, Rules } from 'src/app/jscaip/Rules';
+import { RulesFailure } from 'src/app/jscaip/RulesFailure';
+import { assert } from 'src/app/utils/assert';
+import { MGPFallible } from 'src/app/utils/MGPFallible';
+import { MGPMap } from 'src/app/utils/MGPMap';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { MartianChessMove, MartianChessMoveFailure } from './MartianChessMove';
+import { MartianChessPiece, MartianChessState } from './MartianChessState';
+
+export class MartianChessRulesFailure {
+
+    public static readonly MUST_CHOOSE_PIECE_FROM_YOUR_TERRITORY: () => string = () => $localize`Choose a piece from your territory to move it`;
```

Comment:
Proposition: "You should pick a piece from your side of the board in order to move it" 
Aussi: . manquant

Reply:
Les erreurs de move sont toasté, les toast on les termine par des points ?
Aussi, il fait partie des non toastable celui-là, mais peut être que ça ça doit changer, genre, en cas de premier click qui est chez l'ennemi, toaster celui-là ?

Reply:
Yep, tout se termine par des points sauf les titres (facile à tester: ouvrir n'importe quel jeu et le faire toaster, je confirme donc les points).
Ah tiens oui d'ailleurs justement: si je clique sur une pièce sur le territoire ennemi, rien n'est toasté. Ça serait pas préférable de le toaster du coup ? Ça m'est arrivé quelques fois d'essayer de jouer avec Clair en tant que premier joueur (car clair est en bas ! contrairement au standard établi par TOUS les autres jeux à territoire, i.e., uniquement abalone et epaminondas)

Reply:
Feel free to ignore les commentaires précédents qui reposent cette même question :D !

Je commence l'uniformisation des . pour les non-titres, et celle des clic ennemis qui toastent

Reply:
Instructif: 
1. Encapsule: "you must pick your piece amongst the remaining ones" quand on clic les pièces ennemies
2. LinesOfAction, Sahara, Siam, Yinsh: Y a surtout des corrections négatives "vous pouvez pas sélectionner une pièce de l'ennemi"  mais sur ceux ci c'est positif (vous devez choisir une de vos pièces)
3. Quarto, Reversi: ne disent rien quand on clique sur une case occupée
4. Siam: il t'autorise a faire un premier clic sur une case vide, à réparer

Reply:
je pense que "clic sur une case vide" pour un jeu où on doit cliquer sur des pièces, ça a du sens de ne rien dire. Mais sinon 1-2 est le bon comportement oui, non ?

Reply:
Oui, positif should prevail, c'est Gandalf il a dit!
Mais là pour le 1 je voulais dire qu'outre que ça doit être positif, il devrais surtout dire "prends dans *TA* réserve" plutôt "prend quelque chose qui reste", c'qui est non seulement flou dans la formulation et imprécis car les pièce adverses "sont restantes" :D

C'est plutôt petit, j'ajoute ça a ce ticket ou tu préfère qu'on finisse ces jeux vite :p ?

Reply:
Je préfère qu'on tickettise plus plutôt que d'avoir des tickets qui s'étalent sur des mois avec moultes mini-tickets faits dedans :D Quitte à un jour faire une PR multi-mini-tickets

Reply:
ticketté!